docs/recipes/puppeteer.md

```diff
@@ -0,0 +1,58 @@
+# Testing web apps using Puppeteer
+
+## Dependencies
+
+- [Puppeteer](https://github.com/GoogleChrome/puppeteer)
+	- `npm i --save-dev puppeteer`
+
+## Setup
+
+The first step is setting up a helper to configure the environment:
+
+`./test/helpers/setup.js`
+
+```js
+const puppeteer = require('puppeteer');
+
+module.exports = (test, url) => {
+  test.before(async t => {
+    t.context.browser = await puppeteer.launch();
+    t.context.page = await t.context.browser.newPage();
+    await t.context.page.goto(url);
```

Comment:
Sharing the browser/page between tests might cause unexpected issues because all tests will run in parallel. What I usually do in my ava/puppeteer tests is to start a separate browser instance for each test.

```js
async function withBrowserPage(fn) {
    const browser = await puppeteer.launch();
    const page = browser.newPage();
    try {
        await fn(page);
    } finally {
        await page.close();
        await browser.close();
    }
}

test('foo', t => {
    return withBrowserPage(page => {
         page.goto('https://google.com');
         t.is(await page.title(), 'Google');
    });
});

test('bar', t => {
    return withBrowserPage(page => {
         page.goto('https://bing.com');
         t.is(await page.title(), 'Bing');
    });
});
```